web_poet/overrides.py

```diff
@@ -0,0 +1,146 @@
+    """
+    if value is None:
+        return []
+    if isinstance(value, str):
+        return [value]
+    return list(value)
+
+
+def handle_urls(include: Union[str, Iterable[str]],
+                overrides: Callable,
+                *,
+                exclude: Union[str, Iterable[str], None] = None,
+                priority: int = 500,
+                namespace: str = "",
+                **kwargs
+                ):
+    """
+    Class decorator that indicates that the decorated Page Object should be used instead of the overridden one
+    for a particular set the URLs.
+
+    Which Page Object is overridden is determined by the ``overrides`` parameter.
+
+    Over which URLs the override happens is determined by the ``include``, ``exclude`` and ``priority`` parameters.
+    See the documentation of the `url-matcher <https://url-matcher.readthedocs.io/>`_
+    package for more information about them.
+
+    Different namespaces can be used to create different groups of annotations. The default namespace is the empty
+    string.
+
+    For the example, the following Page Object is decorated with the ``handle_urls`` decorator:
+
+    .. code-block:: python
+
+        @handle_urls("example.com", overrides=ProductPageObject)
+        class ExampleComProductPage(ItemPage):
+            ...
+
+    The annotation indicates that the ``ExampleComProductPage`` Page Object should be used
+    instead of the ``ProductPageObject`` Page Object for all the URLs whose top level domain is ``example.com``.
+
+    Any extra parameters are stored as meta information that can be later used.
+
+    :param include: Defines the URLs that should be handled by the overridden Page Object.
+    :param overrides: The Page Object that should be replaced by the annotated one.
+    :param exclude: Defines URLs over which the override should not happen.
+    :param priority: The resolution priority in case of conflicting annotations.
+    """
+
+    def wrapper(cls):
+        module = sys.modules[cls.__module__]
+        if not hasattr(module, OVERRIDES_NAMESPACES_KEY):
+            setattr(module, OVERRIDES_NAMESPACES_KEY, {})
+
+        handle_urls_dict = getattr(module, OVERRIDES_NAMESPACES_KEY)
+        spec = HandleUrlsSpec(
+            patterns=Patterns(
+                include=_as_list(include),
+                exclude=_as_list(exclude),
+                priority=priority),
+            overrides=overrides,
+            meta=kwargs,
+        )
+        namespace_dict = handle_urls_dict.setdefault(namespace, {})
+        if cls not in namespace_dict:
+            # If it was already defined, we don't want to override it
+            namespace_dict[cls] = spec
+        return cls
+
+    return wrapper
+
+
+def walk_modules(module: str) -> Iterable:
+    """
+    Return all modules from a module recursively. Note that this will import all the modules and submodules.
+    It returns the provided module as well.
+    """
+    def onerror(mod):
```

Comment:
It mentions `mod` from the outer scope. Is there a possibility of an error here? Do we even need an argument here instead of `_`?

Reply:
Good point! updated this on https://github.com/scrapinghub/web-poet/pull/16/commits/f5cffefd521e17c419ef2db004d0fbc717417aa8.

Reply:
Good point! updated this on https://github.com/scrapinghub/web-poet/pull/16/commits/f5cffefd521e17c419ef2db004d0fbc717417aa8.

Reply:
Good point! updated this on https://github.com/scrapinghub/web-poet/pull/16/commits/f5cffefd521e17c419ef2db004d0fbc717417aa8.